src/com/etendoerp/sequences/SequenceDatabaseUtils.java

```diff
@@ -86,9 +90,23 @@ public static BaseOBObject getFromClass(Class<? extends BaseOBObject> cls, Seque
     }
 
     public static BaseOBObject searchSequence(Class<? extends BaseOBObject> cls, String whereClause, Map<String, Object> parameters) {
-        final OBQuery<BaseOBObject> qry = OBDal.getInstance().createQuery((Class<BaseOBObject>) cls,whereClause);
+        final OBQuery<BaseOBObject> qry = OBDal.getInstance().createQuery((Class<BaseOBObject>) cls, whereClause);
         qry.setNamedParameters(parameters);
+        List<BaseOBObject> listOfSequences = qry.list();
+        if (listOfSequences.size() > 1) {
+            throw new OBException(getNameOfADSequencesOnConflicts(parameters));
+        }
         return qry.uniqueResult();
```

Comment:
modify this, if you have the list you can return listOfSequences.get(0) check if not empty, because if you execute the uniqueResult execute the query in database twice

Reply:
If it has more than one element, the if evaluates to true and returns exception. Else, return unique result..